src/ui/public/courier/data_source/search_source.js

```diff
@@ -288,5 +292,16 @@ export function SearchSourceProvider(Promise, Private, config) {
     }
   };
 
+  SearchSource.prototype.clone = function () {
+    const clone = new SearchSource(this.toString());
+    clone.set('index', this.get('index'));
```

Comment:
The `.toString()` is serializing the `index` property, and losing some of the internal FieldList/Field classes, this is essentially working around this quirk.

It'd be possible to do `const clone = new SearchSource(this.toJSON())`, but when doing this the `SearchSource` constructor then uses `_.cloneDeep` and the same thing happens.

I also thought about using the `SearchSource.set` function to set all the state based on the current state, but in that situation we lose the `deepClone` functionality, and could possible be mutating stuff we didn't expect to.

I defaulted to this approach, even though I really don't like it and I'm open to alternatives.